core/app/models/spree/shipment.rb

```diff
@@ -186,7 +186,7 @@ def inventory_units_for_item(line_item, variant = nil)
     end
 
     def item_cost
-      line_items.map(&:total).sum
+      manifest.map { |m| (m.line_item.price + (m.line_item.adjustment_total / m.line_item.quantity)) * m.quantity }.sum
```

Comment:
I confess it was hard for me to understand this line. I think that expanding the block to multiple lines and adding a temporary variable may improve understandability also for other people that don't have much confidence with this part of Solidus, something like this:

```ruby
      manifest.map do |m|
        adjusted_cost = m.line_item.price + m.line_item.adjustment_total / m.line_item.quantity
        adjusted_cost * m.quantity
      end.sum
```